modules/assim.sequential/R/sda.enkf_refactored.R

```diff
@@ -291,8 +292,13 @@ sda.enkf <- function(settings,
       #---------------- setting up the restart argument
       
       if(exists('new.state')){ #Has the analysis been run? Yes, then restart from analysis.
-        restart.arg<-list(runid = run.id, 
-                          start.time = lubridate::ymd_hms(obs.times[t - 1], truncated = 3),
+       
+        if(t == 2){start.time = lubridate::ymd_hms(settings$run$start.date, truncated = 3)}else
+          if(t != 2){start.time = lubridate::ymd_hms(obs.times[t - 1], truncated = 3)}
```

Comment:
Suggested revision for clarity and style.

```suggestion
        if (t == 2) {
          start.time = lubridate::ymd_hms(settings$run$start.date, truncated = 3)
        } else {
          start.time = lubridate::ymd_hms(obs.times[t - 1], truncated = 3)
        }
```